src/qibolab/compilers/default.py

```diff
@@ -66,3 +66,15 @@ def measurement_rule(gate, platform):
         MZ_pulse = platform.create_MZ_pulse(qubit, start=0)
         sequence.add(MZ_pulse)
     return sequence, {}
+
+
+def align_rule(gate, platform):
+    """Align gate to hold/delay execution of the circuit on the specified qubits."""
+    sequence = PulseSequence()
+    if gate.delay == 0:
+        return sequence, {}
+    for qubit in gate.target_qubits:
+        delay_pulse = platform.create_qubit_drive_pulse(qubit, start=0, duration=gate.delay)
+        delay_pulse.amplitude = delay_pulse.frequency = 0
+        sequence.add(delay_pulse)
```

Comment:
The issue with this solution is that, if the drivers are working properly, it will be sent to the instruments in a suboptimal way. For example
```py
gates.RX(0, theta=math.pi)
gates.Align(0, delay=5000)
gates.M(0)
```
will be translated to
```py
rx_pulse = DrivePulse(start=0, amplitude=..., duration=...)
delay_pulse = DrivePulse(start=rx_pulse.finish, amplitude=0, duration=delay)
measurement = ReadoutPulse(start=delay_pulse.finish, amplitude=..., duration=...)
```
and the instrument may try to synthesize a long (here 5000ns) zero waveform which may lead to memory issues (this may be driver dependent, @rodolfocarobene @Jacfomg correct me if I am wrong).

Ideally we would like this to be translated to
```py
rx_pulse = DrivePulse(start=0, amplitude=..., duration=...)
measurement = ReadoutPulse(start=delay_pulse.finish + delay, amplitude=..., duration=...)
```
which should be easier for the drivers to interpret and sends only two pulses with the appropriate delay between them. This may need a small refactoring in the compiler.

Reply:
You are right, this will add limitations due to memory issues regarding Zurich at least.

Reply:
With the rfsoc too... Maybe it could be transpiled as an offset added to the start time to all the subsequent pulses

Reply:
Good points. Having the delayed `Align` gate set a flag to offset the next pulse would indeed be smoother (and was my first thought of implementation), but we'd still need to handle the situation where the delayed `Align` gate is the last gate in the circuit.

Maybe I could add a new pulse type (e.g. `EmptyPulse` or `DummyPulse`), which is skipped by the driver, serving only to exist to ease offsetting the time for the next pulse?

Reply:
> By last do you mean the measurement ?

I mean the `Align` gate being the final gate acting on that qubit in the circuit (assuming no further measurement is to be made on that specific qubit, if that even makes sense). Saying that, I'm still in favour of using a dummy pulse type compared to refactoring the existing compiler rules.

Reply:
> I mean the `Align` gate being the final gate acting on that qubit in the circuit (assuming no further measurement is to be made on that specific qubit, if that even makes sense). 

Is there any potential application of this? If the `Align` does nothing other than delaying the next gate, then I do not see the point of using it as the final gate, as it there would be no "next gate" to be affected. So it would be the same with not using the `Align` at all.

> Saying that, I'm still in favour of using a dummy pulse type compared to refactoring the existing compiler rules.

Either solution is fine with me, as long as the final instruction that is sent to the devices is to wait, and not play a very long zero-amplitude sequence.

The advantage of changing the compiler to implement `Align` by shifting the start time subsequent pulses is that existing drivers (modulo bugs) already know how to play such delays. If we introduce a new dummy pulse, all drivers would need to be updated to know what to do with it. This would probably need more work than changing the compiler. At the end we should go with what would be better for the long term.

Reply:
> Is there any potential application of this? If the `Align` does nothing other than delaying the next gate, then I do not see the point of using it as the final gate, as it there would be no "next gate" to be affected. So it would be the same with not using the `Align` at all.

One reason I can currently think of to pad the circuit runtime is to extend the relaxation time. This is still a thing the user might want to do (for testing purposes, at least), and so I'd rather not block it if we don't have to.

> The advantage of changing the compiler to implement `Align` by shifting the start time subsequent pulses is that existing drivers (modulo bugs) already know how to play such delays. If we introduce a new dummy pulse, all drivers would need to be updated to know what to do with it. This would probably need more work than changing the compiler. At the end we should go with what is better in the long term.

I see. I'm not at all yet familiar with how the drivers are implemented, but perhaps we could use some wrapper class which inherits all properties of the next pulse, except shifting the start and end times by the delay. 

Alternatively, we could filter the aforementioned `DummyPulse` class pulses from being sent to the driver, possibly in the compiler's `Compiler.compile()` method or from wherever the driver is called, but that comes back to my first point that delays should still have an effect at the end of circuits.